### PR TITLE
Rely on `ownerDocument` references for both poppers and reference elements

### PR DESCRIPTION
**Pull request by kasperisager**
This is a continuation of #415 that ensures that both poppers and reference elements rely on `ownerDocument` references whenever needed in order to support poppers and reference elements residing within iframes. Upon further testing I encountered some cases that #415 did not address that this PR does.

### CODE REVIEW

File: packages/popper/src/utils/removeEventListeners.js [hunk 1]
```diff
@@ -5,8 +5,11 @@
  * @private
  */
 export default function removeEventListeners(reference, state) {
+  const ownerDocument = reference.ownerDocument
```
Comment (FezVrasta) [inline packages/popper/src/utils/removeEventListeners.js]: This is repeated in `setupEventListeners`, may we move this to its own module?

Comment (kasperisager) [reply packages/popper/src/utils/removeEventListeners.js]: Certainly!

File: packages/popper/src/utils/getWindow.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/**
+ * Get the window associated with the element
+ * @argument {Element} element
```
Comment (FezVrasta) [inline packages/popper/src/utils/getWindow.js]: missing @returns annotation

Comment (kasperisager) [reply packages/popper/src/utils/getWindow.js]: Added 👌

**Pull request merged**